### PR TITLE
feat: added basic arrival view

### PR DESCRIPTION
**Pull request by Archdoog**

### CODE REVIEW

File: apple/Tests/FerrostarSwiftUITests/Theme/ArrivalViewThemeTests.swift [hunk 1]
```diff
@@ -0,0 +1,15 @@
+import XCTest
+@testable import FerrostarSwiftUI
+
+final class ArrivalViewThemeTests: XCTestCase {
```
Comment (ianthetechie) [inline apple/Tests/FerrostarSwiftUITests/Theme/ArrivalViewThemeTests.swift:4]: I doubt we need this test case TBH; it'll just be one more thing to change if we change our mind on the default constructor args 😂

File: apple/Tests/FerrostarSwiftUITests/Views/ArrivalViewTests.swift [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+            ArrivalView(
+                progress: TripProgress(
+                    distanceToNextManeuver: 5420,
+                    distanceRemaining: 1_420_000,
+                    durationRemaining: 520_800
+                ),
+                distanceFormatter: germanDistanceFormatter,
+                estimatedArrivalFormatter: etaFormatter,
+                durationFormatter: longDurationFormatter,
+                fromDate: referenceDate
```
Comment (ianthetechie) [inline apple/Tests/FerrostarSwiftUITests/Views/ArrivalViewTests.swift:82]: I see that we only partially test localization here. Could we add a German ETA and duration formatter as well? That will help verify the changes I suggested in the defaults.

File: apple/Tests/FerrostarSwiftUITests/Theme/ArrivalViewThemeTests.swift (hunk 1 above)
Comment (Archdoog) [reply apple/Tests/FerrostarSwiftUITests/Theme/ArrivalViewThemeTests.swift:4]: Agree this test is generally useless. The only two reasons I added it anyways are:

1. It's a good additional catch for misconfigurations that could lead to somewhat larger issues (e.g. missing text)
2. It matches what we did for the instructions view. 

We could also just snapshot test the full default ferrostar theme once we've got all the ui built.

Comment (ianthetechie) [reply apple/Tests/FerrostarSwiftUITests/Theme/ArrivalViewThemeTests.swift:4]: > It matches what we did for the instructions view.

😂 I guess feel free to delete that then; I must have missed it.

I'd say visual snapshot image tests of the UI are probably the way to go in issue detection here. Unless we have a specific edge case that we're trying to guard against. And that's probably solvable with other techniques that make illegal state unrepresentable.

Comment (Archdoog) [reply apple/Tests/FerrostarSwiftUITests/Theme/ArrivalViewThemeTests.swift]: It's gone! 😆

File: apple/Tests/FerrostarSwiftUITests/Views/ArrivalViewTests.swift (hunk 2 above)
Comment (Archdoog) [reply apple/Tests/FerrostarSwiftUITests/Views/ArrivalViewTests.swift]: I revised our testing to include formatter tests as well as some locales on the views.

File: apple/Sources/FerrostarSwiftUI/Library/ArrivalFormatters.swift [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+    }
+
+    /// The formatter for duration on the arrival view.
+    public static var durationFormat: DateComponentsFormatter {
+        let formatter = DateComponentsFormatter()
+        formatter.allowedUnits = [.hour, .minute]
+        formatter.unitsStyle = .abbreviated
+        formatter.zeroFormattingBehavior = .dropAll
+        return formatter
+    }
```
[the remark this answers is not in the document]
Comment (Archdoog) [reply apple/Sources/FerrostarSwiftUI/Library/ArrivalFormatters.swift]: While I very much like `Date.RelativeFormatStyle`, I think keeping it concise is important generally. This could be especially true when considering potential translations.

File: apple/Sources/FerrostarSwiftUI/Theme/ArrivalViewTheme.swift [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+public struct DefaultArrivalViewTheme: ArrivalViewTheme {
+    public var style: ArrivalViewStyle = .full
+    public var measurementColor: Color = .primary
+    public var measurementFont: Font = .title2.bold()
+    public var secondaryColor: Color = .secondary
+    public var secondaryFont: Font = .subheadline
+    public var backgroundColor: Color = .init(.systemBackground)
+
+    public init() {
+        //
```
[the remark this answers is not in the document]
Comment (Archdoog) [reply apple/Sources/FerrostarSwiftUI/Theme/ArrivalViewTheme.swift]: Fixed 👍

File: apple/Sources/FerrostarSwiftUI/Resources/Localizable.xcstrings
Comment (ianthetechie) [inline apple/Sources/FerrostarSwiftUI/Resources/Localizable.xcstrings:1]: If we are going to include translations, I'd prefer they be checked by a fluent speaker, so let's drop German for now (the thing I wanted to test wrt localization for this PR was just number and measurement formatting).

Comment (Archdoog) [reply apple/Sources/FerrostarSwiftUI/Resources/Localizable.xcstrings:1]: Removed 👍

**Review by ianthetechie — APPROVED** [review]
